tests/system/src/test/java/test/robot/javafx/scene/DatePickerOnFocusLostTest.java

```diff
@@ -0,0 +1,214 @@
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package test.robot.javafx.scene;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.concurrent.CountDownLatch;
+
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import javafx.application.Application;
+import javafx.application.Platform;
+import javafx.beans.value.ChangeListener;
+import javafx.scene.Scene;
+import javafx.scene.control.Button;
+import javafx.scene.control.DatePicker;
+import javafx.scene.input.KeyCode;
+import javafx.scene.input.MouseButton;
+import javafx.scene.layout.VBox;
+import javafx.scene.robot.Robot;
+import javafx.stage.Stage;
+import javafx.stage.StageStyle;
+import javafx.stage.WindowEvent;
+import javafx.util.StringConverter;
+
+import test.util.Util;
+
+public class DatePickerOnFocusLostTest {
+
+    static CountDownLatch startupLatch = new CountDownLatch(1);
+    static Robot robot;
+    static volatile Stage stage;
+    static volatile Scene scene;
+    static final int SCENE_WIDTH = 200;
+    static final int SCENE_HEIGHT = SCENE_WIDTH;
+    static VBox root;
+    DatePicker datePicker;
+    Button button; // a button to steal the focus
+
+    int onChangeListenerCalled; // counter for onChangeListener callbacks
+
+    String INITIAL_DATE_TEXT = "2015-03-25";
+    // set up a specific time format, coz it is not known what will be used when running the test
+    DateTimeFormatter dateFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+    LocalDate now = LocalDate.parse(INITIAL_DATE_TEXT); // LocalDate.parse() uses the same format
+
+    private void mouseClick(double x, double y) {
+        Util.runAndWait(() -> {
+            robot.mouseMove((int) (scene.getWindow().getX() + scene.getX() + x),
+                    (int) (scene.getWindow().getY() + scene.getY() + y));
+            robot.mousePress(MouseButton.PRIMARY);
+            robot.mouseRelease(MouseButton.PRIMARY);
+        });
+    }
+
+    @Test
+    public void testDatePickerCommitNoTypo() throws Exception {
+        testDatePickerCommit(false);
+    }
+
+    @Test
+    public void testDatePickerCannotCommitTypo() throws Exception {
+        testDatePickerCommit(true);
+    }
+
+    // This test is for verifying a specific behavior.
+    // 1. Click on datePicker
+    // 2. Modify with keystrokes the day of month value, optionally with typo
+    // 3. Click on button to grab the focus and hence attempt to datePicker.commitValue()
+    // 4. Verify that in case of typo, the value was reverted, as well as the editor's text
+    public void testDatePickerCommit(boolean typo) throws Exception {
```

Comment:
you actually can also write a normal JUnit test as far as I get this. Then there is no need for `Util.runAndWait`, `sleep` and so on

Reply:
A headless test is the preferred way to do it, if feasible. If it is, move the test to the `javafx.controls` module.

Reply:
I have removed the robot test, and replaced it by a simple test case in the existing class `DatePickerTest` in `javafx.controls`